src/main/java/skd/command/Parser.java

```diff
@@ -0,0 +1,157 @@
+package skd.command;
+
+import java.util.List;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+/**
+ * The Parser class is responsible for parsing the user input commands
+ * and converting them into corresponding tasks or command types. It can
+ * also process commands to mark tasks as done, unmark them, or delete tasks.
+ */
+public class Parser {
+
+    /**
+     * Parses user's command and returns the same CommandType.
+     *
+     * @param input The input command from the user.
+     * @return The CommandType matching user's input.
+     */
+    public static CommandType parseCommand(String input) {
+        if (input.equalsIgnoreCase("bye")) {
+            return CommandType.BYE;
+        } else if (input.equalsIgnoreCase("list")) {
+            return CommandType.LIST;
+        } else if (input.startsWith("mark")) {
+            return CommandType.MARK;
+        } else if (input.startsWith("unmark")) {
+            return CommandType.UNMARK;
+        } else if (input.startsWith("todo")) {
+            return CommandType.TODO;
+        } else if (input.startsWith("deadline")) {
+            return CommandType.DEADLINE;
+        } else if (input.startsWith("event")) {
+            return CommandType.EVENT;
+        } else if (input.startsWith("delete")) {
+            return CommandType.DELETE;
+        } else if (input.startsWith("find")) {
+            return CommandType.FIND;
+        } else {
+            return CommandType.ETC;
+        }
+    }
+
+    /**
+     * Parses input to create a new Task object based on command type.
+     *
+     * @param input Input command from the user.
+     * @param type  The CommandType to determine type of task.
+     * @return Task object based on input.
+     */
+    public static Task parseAddCommand(String input, CommandType type) {
+        switch (type) {
+        case TODO:
+            String description = input.substring(5).trim();
+            if (description.isEmpty()) {
+                throw new IllegalArgumentException("OOPS!!! The description of a todo cannot be empty.");
+            }
+            return new ToDo(description);
+        case DEADLINE:
+            String[] deadlineParts = input.substring(9).split(" /by ");
+            if (deadlineParts.length < 2 || deadlineParts[0].trim().isEmpty() || deadlineParts[1].trim().isEmpty()) {
+                throw new IllegalArgumentException("OOPS!!! The description or deadline cannot be empty.");
+            }
+            return new Deadline(deadlineParts[0].trim(), deadlineParts[1].trim());
+        case EVENT:
+            String[] eventParts = input.substring(6).split(" /from | /to ");
+            if (eventParts.length < 3 || eventParts[0].trim().isEmpty()
+                    || eventParts[1].trim().isEmpty()
+                    || eventParts[2].trim().isEmpty()) {
```

Comment:
I like how you make an extremely long line shorter by dividing them into 3 parts